### PR TITLE
Support pandas 1.0

### PR DESCRIPTION
**Pull request by ajkerrigan**
Upgrade pandas to 1.0 and fix a few pandas issues with vd 2.x. The high level view:

* Support pandas 1.0
* Add JSON Lines support
* Fix a few errors I've bumped into (with some inspiration from the generic loader testing ideas in #507)
* Add an initial set of tests to cover the fixes

### CODE REVIEW

**Comment by saulpw** [comment]
Cool, @ajkerrigan!  Looking forward to how this shapes up.

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,24 @@
+python-dateutil     # date type
```
Comment (anjakefala) [inline requirements.in]: What's this? :)

Comment (ajkerrigan) [reply requirements.in]: Ah, that's moved over from here as part of pinning:

https://github.com/saulpw/visidata/pull/563/files#diff-b4ef698db8ca845e5845c4618278f29aL1

That whole commit can be left out really, it's too opinionated for this PR. I just figured I was in over my head messing with this stuff and didn't want the extra risk of dependency version changes causing regressions. And it does invite discussion! :)

File: visidata/loaders/_pandas.py [hunk 2]
```diff
@@ -173,13 +215,101 @@ def _selectByILoc(self, mask, selected=True):
...
+    @asyncthread
+    def selectByRegex(self, regex, columns, unselect=False):
+        '''
+        Find rows matching regex in the provided columns. By default, add
+        matching rows to the selection. If unselect is True, remove from the
+        active selection instead.
+        '''
+        import pandas as pd
+        masks = pd.DataFrame([
+            self.df[col.name].astype(str).str.contains(pat=regex, case=False, regex=True)
```
Comment (saulpw) [inline visidata/loaders/_pandas.py]: `case=('I' not in options.regex_flags)` I think.

File: visidata/loaders/_pandas.py [hunk 3]
```diff
@@ -55,42 +61,77 @@ def read_tsv(self, path, **kwargs):
...
-            if not col.startswith("__vd_")  # reserved for internal usage
-        ]
+        if type(df.index) is pd.Int64Index:
+            df.index = pd.RangeIndex(len(df.index))
+        elif type(df.index) is not pd.RangeIndex:
+            df = df.reset_index()
+
+        self.columns = []
+        for col in (c for c in df.columns if not c.startswith("__vd_")):
+            self.addColumn(Column(
```
Comment (anjakefala) [inline visidata/loaders/_pandas.py:117]: ++ for `self.addColumn` =D

File: visidata/loaders/_pandas.py (hunk 2 above)
Comment (ajkerrigan) [reply visidata/loaders/_pandas.py]: Good call 👍

**Comment by anjakefala** [comment]
Thanks for doing all the requested changes @ajkerrigan! =) And for doing all of this wonderful work for the pandas loader. :purple_heart: :purple_heart: I am going to go ahead and merge it in.

**Pull request merged**